### ISSUE TITLE
Activity timeline page is hard to understand

### ISSUE THREAD

**Issue by jiru** [Tatoeba/tatoeba2#2144]
[UX test 2](https://en.wiki.tatoeba.org/articles/show/ux-test-2) and [3](https://en.wiki.tatoeba.org/articles/show/ux-test-3) showed that the [Activity timeline](https://tatoeba.org/contributions/activity_timeline) page is hard to understand. Let’s make it more self-explicit.

I am not familiar with this page actually. I would like to ask what’s the purpose of this page to members who use it (as far as I know @Guybrush88 and @ckjpn).

As mentioned in #1068, the location of this item in the top menu is also questionable. Maybe we could group it with other stats pages?

**Comment by ckjpn** [comment]
Reverse sorted, like it was years ago, might make it easier to understand, since today's date would be at the top, with a lower number.

Adding a title, other than just the month and year, would also likely help.

**Number of Sentences Added**, or something like that.

**Comment by trang** [comment]
The purpose of this page is mostly just to fulfill the curiosity of people who like stats.

**Comment by ckjpn** [comment]
Note that I don't really use it much, even though I occasionally harvest the data from there to put it on this page, so I can get a quick overview of things.

http://tatoeba.ueuo.com/timeline/

This has color-coding, somewhat like what TRANG used to have.

**Comment by ckjpn** [comment]
Note that if we had the last 2 or 3 days listed on the main page, mentioned in another issue, then members could sort of get the feeling of how things were going without needing to go to this page at all.

**Comment by Guybrush88** [comment]
> I would like to ask what’s the purpose of this page to members who use it (as far as I know @Guybrush88 and @ckjpn).

Personally I use it out of curiosity, to have an overall idea on how much Tatoeba is used to add new sentences over time, so from time to time I enjoy seeing how the whole database has grown (and how it is growing) day by day

**Comment by agrodet** [comment]
There's also #1882. I've been thinking of looking at it for a while, but I think there are other issues I should be solving first so I didn't suggest anything. Anyway, since the occasion has come, here's my ideal "Activity timeline page". Note that I didn't feel like building a prototype so the explanation might be lengthy. 

- Remove it from the "Contribute" menu. Place it under "STATS PER LANGUAGE".  
**Remarks:** That makes it necessary to go back to the homepage to access the page. However, the "stats per language" page has the same issue, I think. We could imagine putting them both under the "Community menu". 

- Name it "Activity timeline - Month Year"

- Use a bar chart including **all** contributions. Each type of contributions would have its own color. The number of contributions would be indicated **inside** the bar, with the total on the left, like it is now. The bars would need to be thicker. There are a maximum of 31 days in a month so it's an acceptable design. Using a nice set of colors (below is just for illustration), each bar would be displayed as the following:   
TOTAL | green bar | blue bar | red bar (| purple bar...)

- Define what contributions to display. There's already an issue somewhere about redefining contributions properly (that is not only "sentences added"). That issue could be treated before or after, it doesn't really matter. Maybe it is not necessary to display "Unlinked sentences" and "Deleted sentences" stats on the "Activity timeline" page. 

- Display the bars in chronological order, like they are now, because it's a timeline. The origin should be at the origin. The chart "grows" as time passes.

- Add a legend (again illustration). Green - sentences added. Blue - links created. Red - audio added. 

I've thought of some other "cookie bonuses", but most of them are bad design ideas so I'll keep them for myself and see if I can somehow think of something better :D

**Comment by ckjpn** [comment]
> The purpose of this page is mostly just to fulfill the curiosity of people who like stats.

These are all stats pages. Maybe they should be grouped together somehow.

https://tatoeba.org/eng/stats/sentences_by_language
https://tatoeba.org/eng/stats/native_speakers
https://tatoeba.org/eng/stats/users_languages
https://tatoeba.org/eng/contributions/activity_timeline

**Comment by agrodet** [comment]
I grouped this with #1068 to modify menus as follows

![image](https://user-images.githubusercontent.com/42261966/74695561-bb766980-5237-11ea-963d-5f686f227ba5.png)

Any thoughts ?

-- 

> These are all stats pages. Maybe they should be grouped together somehow.

They are stats of different categories, though. It's doesn't really make sense to add `sentences by language` under `Community`. Also, adding a `Stats` menu isn't very great because space is limited :/ (And also stats are not as important as `Browse` and `Contribute`)

**Comment by jiru** [comment]
@agrodet Well, I don’t think the quick start guide should go inside the menu because it’s mostly useless to people who know it. The purpose of the menu is to give access to the most important/used pages. However, while the quick start guide is an important page to look at when you’re discovering the website, after you got it, there is no need to check it again.

@agrodet However I think it’s a good idea to put the "activity timeline" link inside the stats block. But there is still room for improvement. The way I see it, the current "stats per languages" link is related to "n supported languages" and "n sentences". Similarly, "n contributions today" would be related to "activity timeline". So it might be worth trying to make these relationships visually clearer, for example by putting the "activity timeline" link under "n contributions today". We might as well rephrase it to something more meaningful like "monthly contributions" or "stats per month".

**Comment by ckjpn** [comment]
> ... I don’t think the quick start guide should go inside the menu ...

The main problem with having the Quick Start Guide at the bottom of the page is that it gets overlooked by the people who need it.

Perhaps instead, though, the "Rules and Guidelines" page should be in the menu, since this is something that regular members may want to refer to from time to time.   (I'm just brainstorming.)

Perhaps the whole top navigation menu needs to be reconsidered.

I wonder how many people choose the "Discuss Sentences" menu item, for example.

The "Improve Sentences" menu item, might actually be something that could be merged with the Quick Start Guide.  It likely doesn't get referred to a lot by regular members, either.

The "Native Speakers" menu item sort of implies that you might be able to see a list of the native speakers, but instead your get a page of stats.  I wonder how often regular contributors actually check out this page.

**Comment by agrodet** [comment]
@jiru's comments make sense. Since the topic is the activity timeline, I'll forget the top menu for a while. Following gillux suggestions, I reorganized the Stats block as follows (spacing and stuff can be discussed later on).

<img width="274" alt="image" src="https://user-images.githubusercontent.com/42261966/74811041-feb60280-5333-11ea-85e2-a45f27a36982.png">

The rationale for the ordering is 
1. The core of Tatoeba is its sentences. 
1-bis. We can read "n sentences in k languages" which I think is better than "k languages having n sentences" (Okay, I admit that this one is not a very good argument!)
2. Stats Per Languages closes the "Sentences & Languages" block.
3. Activity timeline is exactly about n contributions today.

I think "today" is the good quantity because
1. Although the activity timeline summarizes the current month contributions, its unit of measure is a day.
2. We have enough daily contributions to not feel ashamed of using that numbers (always several thousands).
3. I think a day is easier to quantify than a month for most people. I also think that a few thousands is easier to imagine than ninety thousands.

**Comment by trang** [comment]
Extracting the relevant part of [my comment in #1068](https://github.com/Tatoeba/tatoeba2/issues/1068#issuecomment-588559796):

----

I think the screenshot above is on the right track.

My only issue is that it feels a bit strange to have two stats (number of sentences + number of languages) associated to one page. It would feel less broken to have each stat followed by a link to some related page.

For the `{x} sentences`, I have two ideas of related page:
- the [random sentence](https://tatoeba.org/eng/sentences/show/random)
- or the [advanced search](https://tatoeba.org/eng/sentences/advanced_search)

Not sure which one is best...

```
{n} sentences
Random sentence >
```
or
```
{n} sentences
Search sentences >
```

**Comment by ckjpn** [comment]
I like this idea:  https://user-images.githubusercontent.com/42261966/74811041-feb60280-5333-11ea-85e2-a45f27a36982.png

> 468 contributions today

Perhaps the following might be better.
* contributions => sentences
* today => since 0:00 UTC

Like this:

468 sentences since 0:00 UTC
3,141 sentences yesterday.

I still think it would be a good idea to add a "yesterday's" number on the main page, too.  This gives both regular contributors  and new contributors better idea of what's going on without needing to click to the timeline every day.

468 sentences since 0:00 UTC
468 sentences since 0:00 UTC

**Comment by ckjpn** [comment]
> My only issue is that it feels a bit strange to have two stats (number of sentences + number of languages) associated to one page.

Both of these stats, though, can be found on the "Stats per Language" page, so I can understand why he suggested this.

**Comment by agrodet** [comment]
I only suggested this screenshot because it was the first part of reworking the Activity timeline page, but I feel like the discussion is going in another direction. Please allow me to redirect you to #2156 where we can focus on the stats block, and keep this issue focused on the activity timeline page.

**Comment by agrodet** [comment]
@jiru If you want to take care of that yourself, stop me right here. Otherwise, I have started to look at the code, and I have several questions concerning what would be the best way to implement the points 3 and 4 I mentioned [above](https://github.com/Tatoeba/tatoeba2/issues/2144#issuecomment-586596827). I know that we didn't decide how the page will be modified, but the number of types of contributions to be considered is to be increased, as we mentioned several times in different places. So anyway, I had a look at it ^^ (and also it would help me building prototypes)

As far as I understand, currently, there is only one behavior concerning the `contributions_stats` table:
- when `type='sentence' & action='insert'` enters the `contributions` table, the corresponding `type='sentence' & action='insert'` line's `sentences` in `contributions_stats` is increased (or the line is added).

So I have several questions: 
1. Should we consider doing the same when `type='link' & action='insert'` enters `contributions`? A link consists of two `contributions` records, but I think the line in `contributions_stats` should be only increased by one. 
2. Should we consider doing the same when `type='link' & action='delete'` enters `contributions`? An unlink consists of two `contributions` records, but I think the line in `contributions_stats` should be only increased by one. 
3. Should we consider doing the same when `type='sentence' & action='delete'` enters `contributions`?
4. (Bonus) Actually I think the best is to split 3. into two cases: when the delete relates to a sentence added today, in `contributions_stats`, the `type='sentence' & action='insert'` line for today should be decreased. Otherwise, increase the related line as mentioned in 3.

I think added the delete line is not a problem (negative number of contributions, etc.) if we correctly label things for what they are. ("10 sentences deleted" = "10 contributions", "20 'sentence' & 'insert' = "20 sentences added", etc.) 

From my point of view, having all these lines in the `contributions_stats` table will help us increase several pages, i.e. the activity timeline and the profile page. However, I surely lack a view of the whole picture, so I'd be really glad if you could give me some guidance ^^

**Comment by jiru** [comment]
@agrodet please go ahead and make a new PR with your ideas. My answer for all your questions is yes.

I don’t think you lack a view of the whole picture. As you figured out already, the `contributions` table doesn’t store exactly one contribution per record, so we currently have to work around this. You could show the numbers of added sentences, removed sentences, added link pairs and removed link pairs as different colors on the stats bars and add a legend.

Strictly speaking, I think "one contribution" should just match one actual user action like "add an original sentence", "add a translation", "link two sentences" etc. However, since we can’t get this information from the `contributions` table, we might as well avoid counting contributions, and just say "these are the contributions: n new sentences, m new link pairs, etc."

I had the idea of changing the schema of the `contributions` table in #1633 to solve that very problem, but that’s another story.

I can definitely give your further guidance, just ask me if you need and we can schedule a call.

**Comment by Guybrush88** [comment]
> we might as well avoid counting contributions, and just say "these are the contributions: n new sentences, m new link pairs, etc.

maybe, at least in my opinion, this could provide users way more useful and detailed stats, rather than just saying "n contributions in overall". This could also give a better view on how users actually contribute, if devs might want to improve the user experience

**Comment by agrodet** [comment]
Things are going well. I widened the range of contributions logged in the `contributions_stats` table as discussed. Here is what my activity timeline page looks like 

![image](https://user-images.githubusercontent.com/42261966/75778704-a81cdf80-5d9b-11ea-95d2-1964fcbe4cca.png)

Any feedback about the U.I. is appreciated.

**Comment by ckjpn** [comment]
I'm not sure it's a good idea, but one idea to consider would be to put each of these stats on a different line.

2020/03/03
new sentences: 22
new links: 7

Perhaps the unlinked and deleted sentences aren't really needed, but they could be added in the same way.

**Comment by jiru** [comment]
@agrodet It looks nice!

I think you can remove the number on the left, it doesn’t make sense to add up all the numbers.

How about prefixing the numbers with plus and minus to better show it’s an addition or a deletion?

How about some description text like "Statistics about the number of contributions over this month." or "over the month of {month}, {year}."

**Comment by ckjpn** [comment]
The + or - idea is a good one, I think.

Instead of "number of linked sentences", perhaps "number of new links between sentence created".  I assume that's what's actually being counted, since these links could be between 2 older sentences.

"Number Links Created" might a more concise and still understandable.

To be consistent.

Number of Sentences Added
Number of Links Removed
Number of Sentences Deleted.

Or, more concise, drop the phrase "Number of", since they are obviously numbers.
